### DISCUSSION TITLE
Is it possible to detect when an HTTP/1.1 request is being sent on an existing connection versus opening a new connection?

### DISCUSSION THREAD

**Discussion by declanvk** [hyperium/hyper#3139] [Q&A] [answered]
Is it possible to detect when an HTTP/1.1 request is being sent on an existing connection versus opening a new connection?

I'm interested in monitoring some latency numbers for an HTTP client that I have, and I want to compare the latencies for new connections versus existing connections.

**Comment by seanmonstar** [comment, marked as answer]
Assuming you are referring to v0.14.x, and the pooling `hyper::Client`, then: perhaps a couple things that likely will be different in 1.0.

- If you're using `tracing`, you could look at the events hyper outputs. I believe it has different messages for re-using a connection versus establishing a new one. Those messages are not part of a stable API, though. As in, they could conceivably be changed in new versions to make the wording clearer, or something.
- If using `HttpConnector` (or a TLS connector that wraps it), you could probably check the [HttpInfo](https://docs.rs/hyper/0.14.*/hyper/client/connect/struct.HttpInfo.html) on the `Response` and peek at the local addr to see if the port is new. But then again, once a socket is closed and the kernel's CLOSE_WAIT time has passed, it could assign that port to a new connection...

**Reply by declanvk** [reply]
For 
> Those messages are part of a stable API, though

Would that be unstable instead?

Thanks for responding, this helps a lot.

**Reply by seanmonstar** [reply]
Correct, typo. I've edited the sentence.
